homeassistant/components/media_player/yamaha.py

```diff
@@ -81,19 +91,49 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         receivers = rxv.RXV(ctrl_url, name).zone_controllers()
 
     for receiver in receivers:
-        if receiver.zone not in zone_ignore:
-            hass.data[KNOWN].add(receiver.ctrl_url)
-            add_devices([
-                YamahaDevice(name, receiver, source_ignore, source_names)
-            ], True)
+        if receiver.zone in zone_ignore:
+            continue
+
+        device = YamahaDevice(name, receiver, source_ignore, source_names)
+
+        # Only add device if it's not already added
+        if device.unique_id not in hass.data[DATA_YAMAHA]:
+            hass.data[DATA_YAMAHA][device.unique_id] = device
+            add_devices([device])
+        else:
+            _LOGGER.debug('Ignoring duplicate receiver %s', name)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    def service_handle(service):
+        """Handle for services."""
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+
+        if service.service != SERVICE_ENABLE_OUTPUT:
+            return
+
+        devices = [device for device in hass.data[DATA_YAMAHA].values()
+                   if device.entity_id in entity_ids]
+
+        for device in devices:
+            port = service.data.get(ATTR_PORT)
+            enabled = service.data.get(ATTR_ENABLED)
+
+            device.enable_output(port, enabled)
+            device.schedule_update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_ENABLE_OUTPUT, service_handle,
+        descriptions.get(SERVICE_ENABLE_OUTPUT), schema=ENABLE_OUTPUT_SCHEMA)
```

Comment:
Remove the descriptions argument. The signature of this method has changed after a core refactor. See above about lazy loading of service descriptions.